zeus/optimizer/batch_size/server/batch_size_state/models.py

```diff
@@ -0,0 +1,245 @@
+        frozen = True
+
+
+class ExplorationsPerBs(BatchSizeBase):
+    """Model representing all explorations performed for that batch size and job. Immutable after it's created.
+
+    Attributes:
+        explorations (list[ExplorationStateModel]): List of exploration states for this batch size and job id.
+    """
+
+    explorations: list[ExplorationStateModel]
+
+    class Config:
+        """Model configuration.
+
+        Make it immutable after it's created.
+        """
+
+        frozen = True
+
+    @root_validator(skip_on_failure=True)
+    def _check_explorations(cls, values: dict[str, Any]) -> dict[str, Any]:
+        """Validate the consistency of explorations.
+
+        We check if
+            - job Id and batch size are consistent across all elements in the explorations.
+            - round number is in descending order without any gaps between.
+        """
+        bs: int = values["batch_size"]
+        job_id: UUID = values["job_id"]
+        exps: list[ExplorationStateModel] = values["explorations"]
+        exps = sorted(exps, key=lambda exp: exp.round_number, reverse=True)
+
+        round_number = -1
+        for exp in exps:
+            if job_id != exp.job_id:
+                raise ValueError(
+                    f"job_id doesn't correspond with explorations: {job_id} != {exp.job_id}"
+                )
+            if bs != exp.batch_size:
+                raise ValueError(
+                    f"Batch size doesn't correspond with explorations: {bs} != {exp.batch_size}"
+                )
+            if round_number != -1 and round_number - 1 != exp.round_number:
+                raise ValueError(
+                    f"Round number is not in order. Should be ordered in desc without any gaps: Expecting {round_number - 1} but got {exp.round_number}"
+                )
+            round_number = exp.round_number
+
+        return values
+
+
+class MeasurementsPerBs(BatchSizeBase):
+    """Model representing all measurements we observed per batch size and job.
+
+    Attributes:
+        measurements (list[MeasurementOfBs]): List of measurements per batch size.
+    """
+
+    measurements: list[MeasurementOfBs]
+
+    @root_validator(skip_on_failure=True)
+    def _check_explorations(cls, values: dict[str, Any]) -> dict[str, Any]:
+        """Validate if job_id and bs are consistent across all items in measurements."""
+        bs: int = values["batch_size"]
+        job_id: UUID = values["job_id"]
+        ms: list[MeasurementOfBs] = values["measurements"]
+
+        for m in ms:
+            if job_id != m.job_id:
+                raise ValueError(
+                    f"job_id doesn't correspond with explorations: {job_id} != {m.job_id}"
+                )
+            if bs != m.batch_size:
+                raise ValueError(
+                    f"Batch size doesn't correspond with explorations: {bs} != {m.batch_size}"
+                )
+
+        return values
+
+
+class ExplorationsPerJob(BaseModel):
```

Comment:
Not really.  The relationship between schema and model will be
Schema -> Model 
`Job` -> `JobState`
`BatchSize` -> `BatchSizeBase`
`GaussianTsArmState` -> `GaussianTsArmStateModel`
`ExplorationState` -> `ExplorationStateModel`
`Measurement` -> `MeasurementOfBs`

Other than these models, all are helpers. 

Hmm, now I see it is not consistent. I think I can change the name to all end with `Model`, or if there is a better naming.

Reply:
Hmm since the SQLAlchemy classes are used less prevalently in the codebase, perhaps name then with a special suffix and give more concise names to Pydantic models? Does SQLAlchemy call those classes `Schema`? Or a `Table`? Whichever naming convention that it uses.